### DISCUSSION TITLE
Using personal Docker account in slim image to avoid pull rate limit?

### DISCUSSION THREAD

**Discussion by ge-vlad-romanov** [renovatebot/renovate#13799] [Request Help]
I am running Renovatebot via its slim image in a self-hosted capacity on a VM in AWS. The run script essentially just runs the slim image and mounts some config into it:

docker run --rm -it -e RENOVATE_TOKEN -e GITHUB_COM_TOKEN -v /tmp:/tmp -v /var/run/docker.sock:/var/run/docker.sock -v $PWD/config.js:/usr/src/app/config.js  renovate/renovate:slim

On the machine where I am running it from I have done a `docker login` with my own account which is at least able to pull the renovate:slim image when I run the above command (previously it was unable to do even that, again because of the rate limit). However when the slim image is running and tries to pull the required docker images it hits the pull rate limit, e.g.

```js
"artifactErrors": [
         {
           "lockFile": "package-lock.json",
           "stderr": "Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit\n"
         }
       ]
```
       
I am assuming that is because it's using the default docker user which has a rate limit of 100 per 6 hours - so I was wondering if I can pass something into the slim image that would make it use **my** docker user? I have found various options in the docs but I am not sure that they are what I'm looking for.

**Comment by ge-vlad-romanov** [comment]
I have tried using hostRules as defined in the documentation:

```js
hostRules: [
    {
      hostType: 'docker',
      username: '<your-username>',
      password: process.env.DOCKER_HUB_PASSWORD,
    },
  ],
``` 
as well as trying it with an additional matchHost: 'docker.io', I thought that this means that it would do a docker login when pulling its sidecar containers but I still hit the toomanyrequests error. It must not be using my account because I verified that I have just under 200 pull requests remaining. 

I must be doing something wrong as I can't find any other threads with this issue, and looking at this https://github.com/renovatebot/renovate/issues/7125 I assume it has been taken care of. Can anyone give any pointers?

**Reply by viceice** [reply]
Please format code blocks.

You probably need to mount a docker config to renovate container, as renovate won't run login for pulling sidecar docker images.

**Reply by ge-vlad-romanov** [reply]
Sorry - I have formatted them now.

I seem to have temporarily got around the issue by pulling the docker.io/renovate/node docker image on the machine that this is running from, I assume that works because I am mounting the docker socket into the running container. However mounting the docker config seems like the better solution. Would you know how to do that? I assume adding something like this to the docker run command:

`-v /home/.docker/config.json:/home/ubuntu/.docker/config.json` - or should it be mounted somewhere else in the container?

**Reply by viceice** [reply]
Yes, that should work.

**Reply by ge-vlad-romanov** [reply]
Thank you for your responses - I'll give that a try

**Reply by viceice** [reply]
I personally use another `registry` docker container as pull through proxy.

https://docs.renovatebot.com/self-hosted-configuration/#dockerimageprefix

The official docker registry image works fine for this. 😉

**Reply by ge-vlad-romanov** [reply]
So do you run a docker container with the registry image (https://hub.docker.com/_/registry) before you run the renovate image, and point the renovate image at the registry you run?

**Reply by viceice** [reply]
Basically yes, but it's always running, as i use it for other kubernetes pods too. 😉

It's also running with minio S3 storage and Redis cache as two node proxy behind Trafik reverse proxy for TLS termination. But you don't need this all for renovate.

You should cache the registry volume, so it can leverage ca hing too. You can also add upstream credentials there to have lower pull limits.

I also use a second registry to locally cache ghcr.io. so you can basically proxy any upstream docker registry.
